### ISSUE TITLE
mustache output?

### ISSUE THREAD

**Issue by mark-hahn** [goodeggs/teacup#53]
I know I can use teacup to output this mustache (teacup would be useful in more complex templates)

```
render ->
  raw '{{#todos}}'
  li '{{uppercase description}}'
  raw '{{/todos}}'

would output ...

{{#todos}}
<li>{{uppercase description}}</li>
{{/todos}}

```

Is there a better way to do this?  Is there something that could be added to teacup to make this easier and more readable?

**Comment by mark-hahn** [comment]
Just after posting I thought of this.  `must` would know whether it is a tag with a closing tag, or just a helper function, by the presence of a function argument.

```
render ->
  must 'todos', ->
    li must 'uppercase description'
```

I don't really know mustache yet so this idea may need a lot more work, but what do you think?

**Comment by mark-hahn** [comment]
I should mention that this is needed for canjs and other frameworks that require mustache.  I've started a project on rivets, then switched to riot, then switched to ampersand.  I spent two days on each.  They had usability or flakiness problems.  I've had a small amount of exposure to react and I'm not a fan.  I want a tiny framework.

Now canjs looks like the best framework but as I said it requires mustache (or ejs which is just as painful).

I'm in the middle of designing some additions to teacup to output mustache.  I hope to make it much cleaner and friendlier than the example above.  If you don't mind I'll post the specs here for comments.  And then I'll fork and implement.  I'll make it a PR if you end up liking it.

And of course if there is already a way to do this let me know asap before I spend too much effort on it.
